### PR TITLE
Added Goerli Infura Endpoint

### PR DESCRIPTION
**Pull request by fubuloubu**
### What was wrong?
Related to Issue #1229

### How was it fixed?
Copy & paste

NOTE: The Goerli endpoint does not accept unauthenticated requests (you must use your project ID for it to work)

#### Cute Animal Picture

tired kitten

### CODE REVIEW

File: web3/auto/infura/goerli.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+)
+from web3.providers.auto import (
+    load_provider_from_uri,
+)
+
+from .endpoints import (
+    INFURA_GOERLI_DOMAIN,
+    build_infura_url,
+)
+
```
Comment (fubuloubu) [inline web3/auto/infura/goerli.py]:
```suggestion

_headers = build_http_headers()
```

File: web3/auto/infura/goerli.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from web3 import Web3
+from web3.middleware import (
+    geth_poa_middleware,
+)
+from web3.providers.auto import (
+    load_provider_from_uri,
+)
+
+from .endpoints import (
+    INFURA_GOERLI_DOMAIN,
```
Comment (fubuloubu) [inline web3/auto/infura/goerli.py]:
```suggestion
    INFURA_GOERLI_DOMAIN,
    build_http_headers,
```

File: web3/auto/infura/goerli.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    build_infura_url,
+)
+
+_infura_url = build_infura_url(INFURA_GOERLI_DOMAIN)
+
+w3 = Web3(load_provider_from_uri(_infura_url))
```
Comment (fubuloubu) [inline web3/auto/infura/goerli.py]:
```suggestion
w3 = Web3(load_provider_from_uri(_infura_url, _headers))
```

Comment (kclowes) [inline web3/auto/infura/goerli.py]: Then you can pass in the headers here: `w3 = Web3(load_provider_from_uri(_infura_url, _headers))`

Comment (kclowes) [reply web3/auto/infura/goerli.py]: nailed it!

**Comment by kclowes** [comment]
Tests are fixed if you want to pull in the latest master. Then I'll merge! Thank you!

**Pull request merged**